packages/date-picker/src/DatePicker/DatePicker.spec.tsx

```diff
@@ -3001,27 +2987,26 @@ describe('packages/date-picker', () => {
             userEvent.type(monthInput, '7');
             userEvent.type(dayInput, '4');
 
-            userEvent.type(yearInput, '{backspace}{backspace}');
-            expect(yearInput).toHaveValue('20');
+            userEvent.type(yearInput, '{backspace}');
+            expect(yearInput).toHaveValue('');
           });
         });
 
         describe('typing new characters', () => {
-          test('even if the resulting value is valid, keeps the input as-is', async () => {
+          test('updates the value', async () => {
             const { monthInput } = renderDatePicker({});
             userEvent.type(monthInput, '1');
             userEvent.tab();
             await waitFor(() => expect(monthInput).toHaveValue('01'));
             userEvent.type(monthInput, '2');
-            await waitFor(() => expect(monthInput).toHaveValue('01'));
+            await waitFor(() => expect(monthInput).toHaveValue('02'));
           });
 
-          test('if the resulting value is not valid, keeps the input as-is', async () => {
+          test('if the resulting value is not valid, clears the input', async () => {
             const { monthInput } = renderDatePicker({});
-            userEvent.type(monthInput, '6');
-            await waitFor(() => expect(monthInput).toHaveValue('06'));
-            userEvent.type(monthInput, '9');
-            await waitFor(() => expect(monthInput).toHaveValue('06'));
+            userEvent.type(monthInput, '0');
+            userEvent.tab();
+            await waitFor(() => expect(monthInput).toHaveValue(''));
```

Comment:
I think this should test something like month `32` or something like that. Testing `0` is a bit confusing

Reply:
I went with `0` to show that it's an invalid value; also, typing any other value auto formats the value to something valid.

Reply:
But `32` is also an invalid month, so we should test that as well

Reply:
`32` won't clear the input though. It will format `3` to `03` then auto focus the next segment, populating that segment with `2`. But I can add a test to test this specific behavior.